integration/spark2/src/main/scala/org/apache/carbondata/spark/rdd/CarbonTableCompactor.scala

```diff
@@ -374,17 +365,32 @@ class CarbonTableCompactor(carbonLoadModel: CarbonLoadModel,
       sparkSession: SparkSession,
       carbonLoadModel: CarbonLoadModel,
       carbonMergerMapping: CarbonMergerMapping): Array[(String, Boolean)] = {
+    val carbonTable = carbonLoadModel.getCarbonDataLoadSchema.getCarbonTable
     val splits = splitsOfSegments(
       sparkSession,
-      carbonLoadModel.getCarbonDataLoadSchema.getCarbonTable,
+      carbonTable,
       carbonMergerMapping.validSegments)
-    val dataFrame = DataLoadProcessBuilderOnSpark.createInputDataFrame(
-      sparkSession,
-      carbonLoadModel.getCarbonDataLoadSchema.getCarbonTable,
-      splits.asScala)
+    val dataFrame = try {
+      // segments to be compacted are set in the threadset() in carbon session, and unset in the end
```

Comment:
please explain in comment why it is required, not just mentioning what the operation is

Reply:
During custom compaction it might so happen that all the segments might be taken into consideration. To avoid this, segments to be considered and set, are explicitly mentioned here.

Reply:
please add it in the comment in code